tests/integration/conftest.py

```diff
@@ -69,3 +69,18 @@ def local_pypiserver() -> Iterator[None]:
             raise RuntimeError("pypiserver didn't start fast enough")
 
         yield
+
+
+def pytest_collection_modifyitems(
+    session: pytest.Session, config: pytest.Config, items: list[pytest.Item]
+) -> None:
+    """Remove redundant tests which don't have to run for the latest code change.
+
+    This function implements a standard pytest hook. Please refer to pytest
+    docs for further information.
+    """
+    skip_mark = pytest.mark.skip(reason="No changes to tested code")
```

Comment:
Initially I planned for an even more verbose "No changes to code tested by this test", but then have forgotten about it. I would have rather kept my current version because `No changes to test` is ambiguous: does it mean that there is nothing to test or does it mean that a test itself was not modified? As for line-wraps well, this is not some code to actually read, so I am inclined to tolerate that, and besides these lines are but tiny specs on modern 40' 160k+ screens.

Reply:
I'd argue that `No changes to tested code` isn't any better in terms of description, but fair enough, it wasn't a deal breaker for me anyway.